pkg/types/installconfig.go

```diff
@@ -182,3 +186,18 @@ type ClusterNetworkEntry struct {
 	// This is the length in bits - so a 9 here will allocate a /23.
 	DeprecatedHostSubnetLength int32 `json:"hostSubnetLength,omitempty"`
 }
+
+// Proxy defines the proxy settings for the cluster.
+type Proxy struct {
+	// httpProxy is the URL of the proxy for HTTP requests.
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+
+	// httpsProxy is the URL of the proxy for HTTPS requests.
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// noProxy is the list of domains for which the proxy should not be used.
+	// +optional
```

Comment:
It might be useful to provide details like https://github.com/openshift/api/pull/335/files#diff-df10631ecb3582affc4c9f77e8049656R33 here too.